src/main/java/seedu/java/util/Storage.java

```diff
@@ -0,0 +1,114 @@
+package seedu.java.util;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Loads and saves the taskList in .txt format.
+ * load()
+ * save()
+ */
+public class Storage {
+    String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads up a txt file and converts it to an Arraylist of Task.
+     * @return ArrayList of Tasks
+     * @throws FileNotFoundException means your text file is missing/corrupted
+     */
+    public ArrayList<Task> load() throws FileNotFoundException {
+        ArrayList<Task> arr = new ArrayList<>();
+        File f = new File(filePath); // create a File for the given file path
+        Scanner s = new Scanner(f); // create a Scanner using the File as the source
+        while (s.hasNextLine()) {
+            String line = s.nextLine();
+            String [] taskArr = line.split("/");
+            Command type = Command.convert(taskArr[0]);
+            boolean isCompleted = toBool(taskArr[1]);
+            String task = taskArr[2];
+            switch (type) {
+            case TODO:
+                try {
+                    arr.add(new Todo(task, isCompleted));
+                } catch (Exception e) {
+                    System.out.println("load fail todo");
+                }
+                break;
+            case DEADLINE:
+                try {
+                    arr.add(new Deadline(task, taskArr[3], isCompleted));
+                } catch (Exception e) {
+                    System.out.println("load fail deadline");
+                }
+                break;
+            case EVENT:
+                try {
+                    arr.add(new Event(task, taskArr[3], isCompleted));
+                } catch (Exception e) {
+                    System.out.println("load fail event");
+                }
+                break;
+            default:
+                System.out.println("Invalid file");
+                break;
+            }
+        }
+        return arr;
+    }
+
+    private static boolean toBool(String com) {
+        if (com.equals("1")) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private String toTxt(ArrayList<Task> taskArr) {
+        String main = "";
+        for (Task x: taskArr) {
+            String type = "";
+            int complete = x.getComplete() ? 1 : 0;
+            String task = x.getTask();
+            if (x instanceof Todo) {
+                type = "T";
+            } else if (x instanceof Deadline) {
```

Comment:
I feel like this set of if/else statements potentially violates Open-Closed Principle, as to add new Task types a completely unrelated class Storage.java will have to be modified

Would there be a better way to do this? (example: forcing all Tasks to implement a method to get its String type "D", "E" or "T" through an abstract method)